BlueDBOnDisk/src/main/java/io/bluedb/disk/file/BlueObjectInput.java

```diff
@@ -0,0 +1,64 @@
+package io.bluedb.disk.file;
+
+import java.io.Closeable;
+import java.io.DataInputStream;
+import java.io.EOFException;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Path;
+
+import io.bluedb.api.exceptions.BlueDbException;
+import io.bluedb.disk.serialization.BlueSerializer;
+
+public class BlueObjectInput<T> implements Closeable {
+
+	private final Path path;
+	private final BlueSerializer serializer;
+	private final DataInputStream dataInputStream;
+			
+	public BlueObjectInput(BlueReadLock<Path> readLock, BlueSerializer serializer) throws BlueDbException {
+		this.path = readLock.getKey();
+		this.serializer = serializer;
+		dataInputStream = openDataInputStream(path.toFile());
+	}
+
+	public T next() throws EOFException, BlueDbException {
+		int objectLength;
+		try {
+			objectLength = dataInputStream.readInt();
+			byte[] buffer = new byte[objectLength];
+			int bytesRead = dataInputStream.read(buffer,0, objectLength);
```

Comment:
I feel like it may not read the entire length which is why their reference code that I copied keeps reading until the amount of bytes have been read or the end of the file is reached.

Reply:
I'm switching to readFully, which should block until the bytes are available.  Let me know if you think that way is problematic.